File: 프로그래머스/Lv.1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var result = 0
    
    for i in left...right {
        var cnt = 0
        for j in 1...Int(Double(i).squareRoot()) {
            if j * j == i {
                cnt += 1
            } else if i % j == 0 {
                cnt += 2
            }
        }
        if cnt % 2 == 0 {
            result += i
        } else {
            result -= i
        }
    }
    
    return result
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        
        for(int i=left; i<=right; i++) {
            int count = count(i);
            if(count % 2 == 0) {
                answer += i;
            } else {
                answer -= i;
            }
        }
        
        return answer;
    }
    
    public int count(int num) {
        int cnt = 0;
        for(int i=1 ; i<= num; i++) {
            if( num % i == 0) cnt++;
        }
        return cnt;
    }
    
    
}